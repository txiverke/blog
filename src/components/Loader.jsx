// @flow

import React from 'react'

import loader from '../assets/imgs/loader.png'

type Props = {
  msg: string
}

const Loader = (props: Props) => 
  <div className="app-loader"> 
    <img src={loader} className="app-loader-icon icon-spinner3" alt="Loader icon"/>
    <h3 className="app-loader-msg">{props.msg}</h3>
  </div>

export default Loader