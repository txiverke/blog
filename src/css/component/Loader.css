.app-loader {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
}
.app-loader-icon {
  font-size: var(--maxSize);
  color: var(--textColor);
  -webkit-animation: app-loader-rotation .6s infinite linear; /* Safari 4+ */
  -moz-animation:    app-loader-rotation .6s infinite linear; /* Fx 5+ */
  -o-animation:      app-loader-rotation .6s infinite linear; /* Opera 12+ */
  animation:         app-loader-rotation .6s infinite linear;
}

.app-loader-msg {
  font-family: Lighter;
  font-size: var(--minSize);
}

@-webkit-keyframes app-loader-rotation {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
@-moz-keyframes app-loader-rotation {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
@-o-keyframes app-loader-rotation {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
@keyframes app-loader-rotation {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}