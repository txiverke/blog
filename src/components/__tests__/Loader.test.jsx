// @flow

import React from 'react'
import { shallow } from 'enzyme'
import Loader from '../Loader'

describe('<Loader />', () => {
  it('should render itself', () => {
    const msg = 'Just a test message'
    const component = shallow(<Loader msg={msg} />)
    expect(component).toMatchSnapshot()
  })
  
})