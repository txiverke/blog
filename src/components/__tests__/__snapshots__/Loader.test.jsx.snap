// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader
    msg="Just a test message"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Loader icon"
          className="app-loader-icon icon-spinner3"
          src="loader.png"
/>,
        <h3
          className="app-loader-msg"
>
          Just a test message
</h3>,
      ],
      "className": "app-loader",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Loader icon",
          "className": "app-loader-icon icon-spinner3",
          "src": "loader.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Just a test message",
          "className": "app-loader-msg",
        },
        "ref": null,
        "rendered": "Just a test message",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Loader icon"
            className="app-loader-icon icon-spinner3"
            src="loader.png"
/>,
          <h3
            className="app-loader-msg"
>
            Just a test message
</h3>,
        ],
        "className": "app-loader",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Loader icon",
            "className": "app-loader-icon icon-spinner3",
            "src": "loader.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Just a test message",
            "className": "app-loader-msg",
          },
          "ref": null,
          "rendered": "Just a test message",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
